#! /usr/bin/env python3

PACKAGE='visp_tracker'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Common
gen.add("angle_appear", double_t, 0, "Maximal angle value to consider an appearing face", 65.0, 0.0, 90.0)
gen.add("angle_disappear", double_t, 0, "Minimal angle value to consider that a face disappear", 75.0, 0.0, 90.0)

# Specific to Klt
gen.add("mask_border", int_t, 0, "Face dead zone size", 5, 0, 50)
gen.add("max_features", int_t, 0, "Maximum number of KLT features", 10000, 0, 30000)
gen.add("window_size", int_t, 0, "Window size arround a KLT point", 5, 3, 7)
gen.add("quality", double_t, 0, "Quality threshold used to consider that a point is valid", 0.05, 0.0001, 0.1)
gen.add("min_distance", double_t, 0, "Min distance between two points", 5.0, 1.0, 50.0)
gen.add("harris", double_t, 0, "Harris value", 0.01, 0, 0.1)
gen.add("size_block", int_t, 0, "Block size", 3, 2, 10)
gen.add("pyramid_lvl", int_t, 0, "Number of levels in the pyramid", 3, 0, 5)


exit(gen.generate(PACKAGE, "tracker_mbt_client", "ModelBasedSettingsKlt"))
